docker/Dockerfile

```diff
@@ -1,23 +1,39 @@
-FROM debian:buster-slim
+# Start from the latest golang base image
+FROM golang:1.13 as builder
+
+# Set the Current Working Directory inside the container
+WORKDIR /app
+
+# Copy go mod and sum files
+COPY go.mod go.sum ./
+
+# Download all dependencies. Dependencies will be cached if the go.mod and go.sum files are not changed
+RUN go mod download
 
-ARG TYKVERSION
+# Copy the source from the current directory to the Working Directory inside the container
+COPY . .
 
-LABEL Description="Tyk Sync docker image" Vendor="Tyk" Version=$TYKVERSION
+# Build the Go app
+RUN CGO_ENABLED=0 go build -o tyk-sync .
+
+
+######## Start a new stage from scratch #######
+FROM debian:buster-slim
 
 RUN apt-get update \
  && apt-get upgrade -y \
  && apt-get install -y --no-install-recommends \
             curl ca-certificates apt-transport-https debian-archive-keyring gnupg \
- && curl -L https://packagecloud.io/tyk/tyk-sync/gpgkey | apt-key add - \
  && apt-get purge -y gnupg \
  && apt-get autoremove -y \
  && rm -rf /root/.cache
 
-RUN echo "deb https://packagecloud.io/tyk/tyk-sync/debian/ buster main" | tee /etc/apt/sources.list.d/tyk_tyk-sync.list \
- && apt-get update \
- && apt-get install -y tyk-sync=$TYKVERSION \
- && rm -rf /var/lib/apt/lists/*
+RUN mkdir -p /opt/tyk-sync
+WORKDIR /opt/tyk-sync
+
+# Copy the Pre-built binary file from the previous stage
+COPY --from=builder /app/tyk-sync /opt/tyk-sync/tyk-sync
 
- WORKDIR /opt/tyk-sync
+ENTRYPOINT ["./tyk-sync"]
 
- ENTRYPOINT ["./tyk-sync"]
+CMD ["--help"]
```

Comment:
Shouldn't the service run by default?

Reply:
i don't think so because it needs a command such as [publish, sync, dump] and it doesn't make sense to assume